### PR TITLE
feat: os input data

### PR DESCRIPTION
**Pull request by drspacemn**
- mimic os test suite
- refactor 
- impl pathfinder trie with starknet api types

### CODE REVIEW

File: src/state/mod.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        let storage = DefaultStorage::default();
+        let statetxn = txn.get_state_reader().unwrap();
+        for (addr, nonce) in diff.address_to_nonce {
+            let class_hash = statetxn
+                .get_class_hash_at(StateNumber::right_after_block(BlockNumber(0)), &addr)
+                .unwrap()
+                .unwrap();
+
+            let contract_commitment =
+                calculate_contract_state_hash(class_hash, patricia_key!("0x0"), nonce);
```
Comment (EvolveArt) [inline src/state/mod.rs:86]: add TODO

File: src/state/mod.rs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+        let txn = writer.begin_rw_txn().unwrap();
+
+        txn.append_state_diff(block_number, state_diff, IndexMap::new())
+            .unwrap();
+
+        let txn = writer.begin_rw_txn().unwrap();
+
+        let mut contract_commitments: Vec<StarkHash> = Vec::new();
+        let mut tree: MerkleTrie<PedersenHash, 251> = MerkleTrie::empty();
```
Comment (EvolveArt) [inline src/state/mod.rs:76]: can't we use the constant defined in the config ?

File: src/state/mod.rs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            nonces: diff.address_to_nonce.clone(),
+            ..StateDiff::default()
+        };
+        // if let Some(dep_classes) = deprecated_class_mapping {
+        //     state_diff.deprecated_declared_classes = dep_classes;
+        // }
+
+        let (_, mut writer) = open_storage(self.db_conf.clone()).unwrap();
+
+        let txn = writer.begin_rw_txn().unwrap();
```
Comment (EvolveArt) [inline src/state/mod.rs:68]: Create issue to remove unwraps

File: tests/snos.rs [hunk 4]
```diff
@@ -1,45 +1,55 @@
...
+#[rstest]
+fn shared_state(initial_state: TestingContext) {
+    let mut shared_state = SharedState::new(
+        initial_state.block_context.chain_id,
+        Box::new(initial_state.state),
+    );
+    let commitment = shared_state.apply_diff(BlockNumber(1));
+
+    assert_eq!(
+        stark_felt!("473010ec333f16b84334f9924912d7a13ce8296b0809c2091563ddfb63011d"),
```
Comment (EvolveArt) [inline tests/snos.rs:41]: where does that come from?

**Review by EvolveArt — APPROVED** [review]
lgtm

File: tests/snos.rs (hunk 4 above)
Comment (drspacemn) [reply tests/snos.rs:41]: this is the equivalent root for the python OS tests, will add comment

**Pull request merged**